[Unity] Merge with tvm/unity plus a few fixes

jwfromm opened the pull request:
Grab a few recent changes off tvm/unity and incorporate some fixes that were preventing all high priority models from running ith layout transformation.

yongwww reviewed (changes requested):
Thanks Yuchen for the work! looking forward to seeing the e2e case works! It would be great to have the relay translator as an independent pr first if there are some further work needed for the bert support.

YuchenJin commented:
> Thanks Yuchen for the work! looking forward to seeing the e2e case works! It would be great to have the relay translator as an independent pr first if there are some further work needed for the bert support.

Thanks @yongwww for the suggestion! I will break down this WIP pr into several pieces and add test cases to each:
- [ ] `ConstantNode` support in codegen and runtime
- [ ] Changes to script/parser to support `ConstantNode`
- [ ] Improve BlockBuilder to avoid generating duplicated PrimFuncs
- [ ] Static Relay -> Relax translator

YuchenJin commented:
Closing this, the subtasks of this PR have been merged, and the translator is implemented in https://github.com/tlc-pack/relax/pull/75. We will keep the BERT dynamic demos in a separate branch because the Relay forward+backward graph is in text format.

sunggg committed:
[Unity][Transform] Introduce data-dependent operation of reshape and its constant folding (#14282)

* FEAT: Support data-dependent operation of reshape

* FEAT: Support constant folding with data-dependent reshape

* fix

* remove empty line

* reflect feedback

* Lift the lowering of tensor_to_shape from builtin to DecomposeCompositeOps pass

* fix and comment

* fix

* add comments

* reflect feedback

* add comment

* fix

spectrometerHBH committed:
[Unity][Transform] Fix AMP tests (#14360)

tests

quic-sanirudh committed:
[Unity] Add support to append relay op attrs in translator (#14356)

This PR adds an argument to the relay to relax translator to append the
relay Op Attrs as function attributes to the generated TIR PrimFuncs

This information is really useful in Relax as the absence of this
prevents us from being able schedule efficiently for ops that are
heavily sensitive to the attributes.

For example, the `groups` attribute to `conv2d` op is needed to
differentiate between regular conv2d and depthwise conv2d.

tqchen committed:
[Unity][WEB] Support async pipeline creation (#14362)

This PR enables async pipeline creation in webgpu module loading.
This will enable us to report progress in shader compilation
and leverage multi-threading on the host to compile shaders.

Commit:
[Unity][Pass] Fix FuseOps error if there is no output of a given group (#14354)

This PR is to fix the FuseOps error if there is no output of a given group,
although the pass `DeadCodeElimination` can solve the problem, it is better
to enhance the robustness of the pass `FuseOps`.

psrivas2 committed:
[Unity][Fix] Infer Layout must support negative axes (#14365)

This PR fixes a bug in Infer Layout for reduction ops in when `axis` has negative indices.

jinhongyii committed:
[Unity] Add More Ops For FX Translator (#14348)

This PR makes 2 changes:
1. Add Relax Op Maximum and Minimum
2. Add translation function for torch function/method silu, to, ones, full, masked_fill_, mean, rsqrt, neg, max in fx translator

Commit:
[Unity][TVMScript] Update GlobalVar `checked_type_` when `emit_te` (#14367)

The current GlobalVar generated by `emit_te` has empty `checked_type_`,
which will fail at the well-formed check, saying

```
Warning: This IR is not well formed: The checked_type_ of Expr I.GlobalVar("add") is nullptr.
```

This PR fixes this issue and enables well-formed checks in parser tests.

Commit:
Merge remote-tracking branch 'tvm/unity' into unity_merge

Commit:
Minor fixes to enable all models with layout transform
